lib/entities/purpose.dart

```diff
@@ -0,0 +1,25 @@
+// Class used to represent a purpose.
+class Purpose implements Comparable<Purpose> {
+  // Purpose ID
+  String? id;
```

Comment:
Could id be not nullable, so we return null or an error if it is not present?

Reply:
we'll keep as nullable for now considering null/nil values can come from native SDKs.